### PR TITLE
Fix libssh2 to use the conan-specified openssl variables.

### PR DESCRIPTION
**Pull request by bemehiser**
libssh2/*

The variables set by Conan openssl in a generated CMake find script don't match those used by libssh2.

Before:
```
-- Looking for EVP_aes_128_ctr
-- Looking for EVP_aes_128_ctr - not found
```

Now:
```
-- Looking for EVP_aes_128_ctr
-- Looking for EVP_aes_128_ctr - found
```

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libssh2/all/patches/0002-openssl.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ 
+-  if(OPENSSL_FOUND)
++  if(OpenSSL_FOUND)
+     set(CRYPTO_BACKEND "OpenSSL")
+     set(CRYPTO_SOURCES openssl.c openssl.h)
+     list(APPEND PRIVATE_COMPILE_DEFINITIONS LIBSSH2_OPENSSL)
+-    list(APPEND PRIVATE_INCLUDE_DIRECTORIES ${OPENSSL_INCLUDE_DIR})
+-    list(APPEND LIBRARIES ${OPENSSL_LIBRARIES})
++    list(APPEND PRIVATE_INCLUDE_DIRECTORIES ${OpenSSL_INCLUDE_DIR})
++    list(APPEND LIBRARIES ${OpenSSL_LIBRARIES})
```
Comment (madebr) [inline recipes/libssh2/all/patches/0002-openssl.patch:17]: This is weird, we're manually creating these variables in the openssl module:

https://github.com/conan-io/conan-center-index/blob/9f038df28e5a77044ac2230673b16465a6c20bbd/recipes/openssl/3.x.x/conanfile.py#L675-L677


https://github.com/conan-io/conan-center-index/blob/9f038df28e5a77044ac2230673b16465a6c20bbd/recipes/openssl/3.x.x/conanfile.py#L692-L694

Do you have an idea why this is not working?

Comment (bemehiser) [reply recipes/libssh2/all/patches/0002-openssl.patch:17]: Yeah, but for some reason they don't show up in the generated cmake find scripts - probably only some other cmake generator configuration.

Comment (madebr) [reply recipes/libssh2/all/patches/0002-openssl.patch:17]: Doesn't the generated `FindOpenSSL.cmake` module include a `conan-official-openssl-variables.cmake` file?

Comment (SpaceIm) [reply recipes/libssh2/all/patches/0002-openssl.patch:17]: It works fine for me without these patches, your openssl recipe is outdated I guess.

**Pull request closed without merging**